distr-otel: add description for all TP extensions (#14)
max-cx/openshift-docs: modules/distr-tracing-otel-config-collector.adoc

[id="basicauth-extension_{context}"]
==== BasicAuth extension

> You can use the BasicAuth extension as an authenticator for HTTP and gGRPC based receivers and exporters.

Reviewer: ```suggestion
You can use the BasicAuth extension as an authenticator for receivers and exporters that are based on the HTTP protocol and the gRPC framework.
```

Author: ```suggestion
You can use the BasicAuth extension as an authenticator for receivers and exporters that are based on the HTTP and the gRPC protocol.
```

        auth:
          authenticator: basicauth/client <6>

    service:
      extensions: [basicauth/server, basicauth/client]
      pipelines:
        traces:
          receivers: [otlp]
          exporters: [otlp]
----
<1> The BasicAuth extension can be configured as a server authenticator that reads credentials from a `.htpasswd` file.
<2> The BasicAuth extension can be configured as a client authenticator to read the credentials from an inline string that consists of environment variables.
<3> The client username is configured as a client authenticator for the BasicAuth extension.
<4> The client password is configured as a client authenticator for the BasicAuth extension.
<5> The authenticator configuration can be assigned to an OTLP receiver.
<6> The authenticator configuration can be assigned to an OTLP exporter.

[id="bearertokenauth-extension_{context}"]

> ==== BearertokenAuth extension

Reviewer: Should it be `BearerTokenAuth`?
```suggestion
==== BearerTokenAuth extension
```

> You can use the BearertokenAuth extension as an authenticator for HTTP and gGRPC based receivers and exporters.

Reviewer: ```suggestion
You can use the BearerTokenAuth extension as an authenticator for receivers and exporters that are based on HTTP and gRPC.
```

> Client authentication and server authentication for the BearertokenAuth extension can be used on the receiver and exporter side in the OpenTelemetry Collector custom resource.

Reviewer: ```suggestion
You can use the OpenTelemetry Collector custom resource to configure client authentication and server authentication for the BearerTokenAuth extension on the receiver and exporter side.
```

<6> Time interval to check the number of failures. Default is `5m`.
<7> The failure number threshold to mark containers as healthy. Default is `5`.

[id="balast-extension_{context}"]
==== Balast extension

Memory Ballast extension enables applications to configure memory ballast for the process.

* Support level: link:https://access.redhat.com/support/offerings/techpreview[Technology Preview]

.OpenTelemetry Collector custom resource with configured Balast extension
[source,yaml]
----
  config: |
    extensions:
      memory_ballast:
        size_mib: 64 <1>
        size_in_percentage: 20 <2>

    receivers:
      otlp:
        protocols:
          http:

    exporters:
      otlp:

    service:
      extensions: [memory_ballast]
      pipelines:
        traces:
          receivers: [otlp]
          exporters: [otlp]
----
<1> Is the memory ballast size, in MiB. Takes higher priority than `size_in_percentage` if both are specified at the same time.
<2> Sets the memory ballast based on the total memory in percentage, value range is 1-100. It is supported in both containerized(eg, docker, k8s) and physical host environments.

[id="zpages-extension_{context}"]

> ==== zPages extension

Reviewer 2: The zPages extension should be as well documented in the troubleshooting page

Author: Do we already have a troubleshooting page for distr?

Reviewer 2: https://docs.openshift.com/container-platform/4.13/distr_tracing/distr_tracing_otel/distr-tracing-otel-using.html

Reviewer 2: https://docs.openshift.com/container-platform/4.13/distr_tracing/distr_tracing_otel/distr-tracing-otel-troubleshooting.html

The zPage extension enables an extension that serves zPages, an HTTP endpoint that provides live data for debugging different components that were properly instrumented for such. All core exporters and receivers provide some zPage instrumentation.

zPages are useful for in-process diagnostics without having to depend on any backend to examine traces or metrics.

* Support level: link:https://access.redhat.com/support/offerings/techpreview[Technology Preview]

.OpenTelemetry Collector custom resource with configured Zpage extension
[source,yaml]
----
  config: |
    extensions:
      zpages:
        endpoint: "localhost:55679"

    receivers:
      otlp:
        protocols:
          http:
    exporters:
      otlp:

    service:
      extensions: [zpages]
      pipelines:
        traces:
          receivers: [otlp]
          exporters: [otlp]
----

> <1> Specifies the HTTP endpoint that serves zPages. Use localhost: to make it available only locally, or ":" to make it available on all network interfaces. Default is `localhost:55679`.

Reviewer 2: the marker is missing in the yaml

* Support level: link:https://access.redhat.com/support/offerings/techpreview[Technology Preview]
* Supported signals: traces, metrics, logs

.OpenTelemetry Collector custom resource with client authentication configured for the OAuth2Client extension
[source,yaml]
----
  config: |
    extensions:
      oauth2client:
        client_id: <client_id> <1>
        client_secret: <client_secret> <2>
        endpoint_params: <3>
          audience: <audience>
        token_url: https://example.com/oauth2/default/v1/token <4>
        scopes: ["api.metrics"] <5>
        # tls settings for the token client
        tls: <6>
          insecure: true <7>
          ca_file: /var/lib/mycert.pem <8>
          cert_file: <cert_file> <9>
          key_file: <key_file> <10>
        timeout: 2s <11>

    receivers:
      otlp:
        protocols:
          http:

    exporters:
      otlp:
        auth:
          authenticator: oauth2client <12>

    service:
      extensions: [oauth2client]
      pipelines:
        traces:
          receivers: [otlp]
          exporters: [otlp]
----

> <1> The client identifier for the OpenTelemetry Collector application.
> <2> The secret that is shared between the client and authorization server.
> <3> You can add further metadata as key-value pairs to be transferred during authentication.
> <4> The token endpoint that is used to retrieve an access token.
> 

Author: ```suggestion
```

          receivers: [spanmetrics]
----
<1>: Defines the flush interval of the generated metrics. Defaults to `15s`.

[id="bearertokenauth-extension_{context}"]
==== BearerTokenAuth extension

> You can use the BearerTokenAuth extension as an authenticator for receivers and exporters that are based on HTTP and gRPC.

Reviewer: ```suggestion
You can use the BearerTokenAuth extension as an authenticator for receivers and exporters that are based on the HTTP protocol and the gRPC framework.
```

Author: ```suggestion
You can use the BearerTokenAuth extension as an authenticator for receivers and exporters that are based on the HTTP and the gRPC protocol.
```

You can use the OpenTelemetry Collector custom resource to configure client authentication and server authentication for the BearerTokenAuth extension on the receiver and exporter side.

* Support level: link:https://access.redhat.com/support/offerings/techpreview[Technology Preview]
* Supported signals: traces, metrics, logs

> .OpenTelemetry Collector custom resource with client and server authentication configured for the BearertokenAuth extension

Reviewer: ```suggestion
.OpenTelemetry Collector custom resource with client and server authentication configured for the BearerTokenAuth extension
```

[source,yaml]
----
  config: |
    extensions:
      bearertokenauth:
        scheme: "Bearer" <1>

>         token: "somerandomtoken" <2>

Reviewer: ```suggestion
        token: "<token>" <2>
```

>         filename: "file-containing.token" <3>

Reviewer: Is `.token` a file xtension? If not, then my suggestion is to use the OpenShift docs' syntax for user-replaced values:
```suggestion
        filename: "<token_file>" <3>
```

    receivers:
      otlp:
        protocols:
          http:
            auth:
              authenticator: bearertokenauth <4>
    exporters:
      otlp:
        auth:
          authenticator: bearertokenauth <5>

    service:
      extensions: [bearertokenauth]
      pipelines:
        traces:
          receivers: [otlp]
          exporters: [otlp]
----

> <1> The BearertokenAuth extension can be configured to send a custom `scheme`, but `Bearer` is used as default.

Reviewer: ```suggestion
<1> You can configure the BearerTokenAuth extension to send a custom `scheme`. The default is `Bearer`.
```

> <2> The BearertokenAuth extension token added as metadata identify a message.

Reviewer: I rephrased it in my suggestion below, but I'm not sure if this is what you meant:
```suggestion
<2> You can add the BearerTokenAuth extension token as metadata to identify a message.
```

> <3> Path to a file that contains a authorization token that is transmitted with every message.

Reviewer: ```suggestion
<3> Path to a file that contains an authorization token that is transmitted with every message.
```

<4> The authenticator configuration can be assigned to an OTLP receiver.

> <5> The authenticator configuration can be assigned to an OTLP exporter.

Reviewer: The docs guidelines ask for active voice (e.g., "You can") rather than passive voice (e.g., "can be") wherever possible:
```suggestion
<4> You can assign the authenticator configuration to an OTLP receiver.
<5> You can assign the authenticator configuration to an OTLP exporter.
```

[id="oauth2client-extension_{context}"]
==== OAuth2Client extension

> You can use the OAuth2Client extension as authenticator on HTTP or gRPC based exporters.

Reviewer: ```suggestion
You can use the OAuth2Client extension as an authenticator for exporters that are based on the HTTP protocol and the gRPC framework.
```

Author: ```suggestion
You can use the OAuth2Client extension as an authenticator for exporters that are based on the HTTP and the gRPC protocol.
```

> Client authentication for the OAuth2Client extension is configured in a separate sections in the OpenTelemetry Collector custom resource.

Reviewer: ```suggestion
Client authentication for the OAuth2Client extension is configured in a separate section in the OpenTelemetry Collector custom resource.
```

* Support level: link:https://access.redhat.com/support/offerings/techpreview[Technology Preview]
* Supported signals: traces, metrics, logs

.OpenTelemetry Collector custom resource with client authentication configured for the OAuth2Client extension
[source,yaml]
----
  config: |
    extensions:
      oauth2client:

>         client_id: someclientid <1>

Reviewer: ```suggestion
        client_id: <client_id> <1>
```

>         client_secret: someclientsecret <2>

Reviewer: ```suggestion
        client_secret: <client_secret> <2>
```

        endpoint_params: <3>

>           audience: someaudience

Reviewer: What is `audience`? Is this self-explanatory to readers? If not, then I suggest adding a callout `<n>` to explain what this is and which values are supported or where those values are taken from if not arbitrarily chosen by the reader.
```suggestion
          audience: <audience>
```

        token_url: https://example.com/oauth2/default/v1/token <4>
        scopes: ["api.metrics"] <5>
        # tls settings for the token client
        tls: <6>
          insecure: true <7>
          ca_file: /var/lib/mycert.pem <8>

>           cert_file: certfile <9>

Reviewer: ```suggestion
          cert_file: <cert_file> <9>
```

>           key_file: keyfile <10>

Reviewer: Is this a path to the key file? If so, let's clarify this like `<path_to_key_file>`.
```suggestion
          key_file: <key_file> <10>
```

>         # timeout for the token client

Reviewer: Let's not keep any comments in the YAML block since we already have the explanations in the callouts below the block (like `<11>` for this).
```suggestion
```

        timeout: 2s <11>

    receivers:
      otlp:
        protocols:
          http:

    exporters:
      otlp:
        auth:

>           authenticator: oauth2client <12>

Reviewer: Is `oauth2client` a value or a placeholder for some value that the user has to enter here?

Author: Thats the extension that was created on top and initlized in the service section.

    service:
      extensions: [oauth2client]
      pipelines:
        traces:
          receivers: [otlp]
          exporters: [otlp]
----
<1> Client identifier for the OpenTelemetry Collector application.
<2> Secret shared between the client and authorization server.
<3> Further metadata which is transferred during authentication. Consisting of key and value.

> <4> Token endpoint used to retrieve an access token using

Reviewer: Could you please check the line with  `<4> Token endpoint used to retrieve an access token using`?
Are some words missing after `using` there?

```suggestion
<1> The client identifier for the OpenTelemetry Collector application.
<2> The secret that is shared between the client and authorization server.
<3> You can add further metadata as key-value pairs to be transferred during authentication.
<4> The token endpoint that is used to retrieve an access token.
```

<1> Client identifier, which is provided by the identity provider.
<2> Confidential key used to authenticate the client to the identity provider.
<3> Further metadata which is transferred during authentication. Consisting of key and value. For example audience specifies the intended audience for the access token. It indicates the recipient of the token.

> <4> URL of the OAuth2 token endpoint, where the Collector will request access tokens.

Reviewer: Seems like there are already callouts `<1>`-`<4>` starting on line 542 above.
Could you please check where these lines come from (maybe previously copied and pasted from another file) and probably delete them?

> <5> Scopes define the specific permissions or access levels requested by the client.

Reviewer: ```suggestion
<5> The scopes define the specific permissions or access levels requested by the client.
```

> <6> Specifies TLS (Transport Layer Security) settings for the token client, which is used to establish a secure connection when requesting tokens.

Reviewer: ```suggestion
<6> The Transport Layer Security (TLS) settings for the token client, which is used to establish a secure connection when requesting tokens.
```

> <7> Flag that when set to true, configures the collector to use an insecure (non-verified) TLS connection when to call the configured token endpoint.

Reviewer: ```suggestion
<7> When set to `true`, configures the Collector to use an insecure or non-verified TLS connection to call the configured token endpoint.
```

> <8> Parameter that specifies the path to a CA (Certificate Authority) file used to verify the server's certificate during the TLS handshake.

Reviewer: ```suggestion
<8> The path to a Certificate Authority (CA) file that is used to verify the server's certificate during the TLS handshake.
```

> <9> Parameter that specifies the path to the client certificate file, which the client may use to authenticate itself to the OAuth2 server if required.

Reviewer: ```suggestion
<9> The path to the client certificate file that the client must use to authenticate itself to the OAuth2 server if required.
```

> <10> Parameter that specifies the path to the client's private key file, which is used in conjunction with the client certificate for authentication if needed.

Reviewer: ```suggestion
<10> The path to the client's private key file that is used with the client certificate if needed for authentication.
```

<11> Sets a timeout for the token client's request.

> <12> The authenticator configuration can be assigned to an OTLP exporter.

Reviewer: ```suggestion
<12> You can assign the authenticator configuration to an OTLP exporter.
```

[id="jaegerremotesampling-extension_{context}"]
==== Jaeger Remote Sampling extension

> This extension allows serving sampling strategies following the Jaeger's remote sampling API. This extension can be configured to proxy requests to a backing remote sampling server, which could potentially be a Jaeger Collector down the pipeline, or a static JSON file from the local file system.

Reviewer: What exactly did you mean by `following Jaeger's remote sampling API`? Did you mean that it happens downstream from (after) Jaeger's remote sampling API or that it happens in compliance with Jaeger's remote sampling API rules/spec?
```suggestion
This extension allows serving sampling strategies after Jaeger's remote sampling API. You can configure this extension to proxy requests to a backing remote sampling server such as a Jaeger collector down the pipeline or to a static JSON file from the local file system.
```

* Support level: link:https://access.redhat.com/support/offerings/techpreview[Technology Preview]

> .OpenTelemetry Collector custom resource with configured Jaeger Remote Sampling extension

Reviewer: ```suggestion
.OpenTelemetry Collector custom resource with a configured Jaeger Remote Sampling extension
```

[source,yaml]
----
  config: |
    extensions:
      jaegerremotesampling:
        source:
          reload_interval: 30s <1>
          remote:
            endpoint: jaeger-collector:14250 <2>
          file: /etc/otelcol/sampling_strategies.json <3>

    receivers:
      otlp:
        protocols:
          http:

    exporters:
      otlp:

    service:
      extensions: [jaegerremotesampling]
      pipelines:
        traces:
          receivers: [otlp]
          exporters: [otlp]
----

> <1> Interval in which the sampling configuration is updated.

Reviewer: ```suggestion
<1> The time interval at which the sampling configuration is updated.
```

> <2> Specifies the endpoint where the Jaeger remote sampling strategy provider can be reached.

Reviewer: ```suggestion
<2> The endpoint for reaching the Jaeger remote sampling strategy provider.
```

> <3> Specifies the path to a local file containing sampling strategy configuration in JSON format.

Reviewer: ```suggestion
<3> The path to a local file that contains a sampling strategy configuration in the JSON format.
```

> .Example of Jaeger Remote Sampling strategy file

Reviewer: ```suggestion
.Example of a Jaeger Remote Sampling strategy file
```

[source,json]
----
{
  "service_strategies": [
    {
      "service": "foo",
      "type": "probabilistic",
      "param": 0.8,
      "operation_strategies": [
        {
          "operation": "op1",
          "type": "probabilistic",
          "param": 0.2
        },
        {
          "operation": "op2",
          "type": "probabilistic",
          "param": 0.4
        }
      ]
    },
    {
      "service": "bar",
      "type": "ratelimiting",
      "param": 5
    }
  ],
  "default_strategy": {
    "type": "probabilistic",
    "param": 0.5,
    "operation_strategies": [
      {
        "operation": "/health",
        "type": "probabilistic",
        "param": 0.0
      },
      {
        "operation": "/metrics",
        "type": "probabilistic",
        "param": 0.0
      }
    ]
  }
}
----

[id="pprof-extension_{context}"]

> ==== PPROF extension

Reviewer: I couldn't find any source where this `PPROF` is unabbreviated (not even in https://github.com/google/pprof/blob/main/doc/README.md), so if possible, we'd better replace `PPROF` in this heading:
```suggestion
==== Performance Profiler extension
```

> Performance Profiler extension enables the golang `net/http/pprof` endpoint. This is typically used by developers to collect performance profiles and investigate issues with the service.

Reviewer: ```suggestion
The Performance Profiler extension enables the Go `net/http/pprof` endpoint. This is typically used by developers to collect performance profiles and investigate issues with the service.
```

* Support level: link:https://access.redhat.com/support/offerings/techpreview[Technology Preview]

> .OpenTelemetry Collector custom resource with configured pprof extension

Reviewer: ```suggestion
.OpenTelemetry Collector custom resource with the configured Performance Profiler extension
```

[source,yaml]
----
  config: |
    extensions:
      pprof:
        endpoint: localhost:1777 <1>
        block_profile_fraction: 0 <2>
        mutex_profile_fraction: 0 <3>
        save_to_file: test.pprof <4>

    receivers:
      otlp:
        protocols:
          http:

    exporters:
      otlp:

    service:
      extensions: [pprof]
      pipelines:
        traces:
          receivers: [otlp]
          exporters: [otlp]
----

> <1> Endpoint in which the pprof will be listening to. Use localhost: to make it available only locally, or ":" to make it available on all network interfaces. Default value is `localhost:1777`.

Reviewer: ```suggestion
<1> The endpoint at which this extension listens. Use `localhost:` to make it available only locally or `":"` to make it available on all network interfaces. The default value is `localhost:1777`.
```

> <2> Fraction of blocking events that are profiled. A value <= 0 disables profiling. See https://golang.org/pkg/runtime/#SetBlockProfileRate for details. Default vlaue is 0.

Reviewer: ```suggestion
<2> Sets a fraction of blocking events to be profiled. To disable profiling, set this to `0` or a negative integer. See the link:https://golang.org/pkg/runtime/#SetBlockProfileRate[documentation] for the `runtime` package. The default value is `0`.
```

> <3> Fraction of mutex contention events that are profiled. A value <= 0 disables profiling. See https://golang.org/pkg/runtime/#SetMutexProfileFraction for details. Default value is 0.

Reviewer: ```suggestion
<3> Set a fraction of mutex contention events to be profiled. To disable profiling, set this to `0` or a negative integer. See the link:https://golang.org/pkg/runtime/#SetMutexProfileFraction[documentation] for the `runtime` package. The default value is `0`.
```

> <4> File name to save the CPU profile to. The profiling starts when the Collector starts and is saved to the file when the Collector is terminated.

Reviewer: ```suggestion
<4> The name of the file in which the CPU profile is to be saved. Profiling starts when the Collector starts. Profiling is saved to the file when the Collector is terminated.
```

[id="healthcheck-extension_{context}"]
==== Health Check extension

> The Health Check extension enables an HTTP url that can be probed to check the status of the OpenTelemetry Collector. This extension can be used as a liveness and/or readiness probe on OpenShift.

Reviewer: ```suggestion
The Health Check extension provides an HTTP URL for checking the status of the OpenTelemetry Collector. You can use this extension as a liveness and readiness probe on OpenShift.
```

* Support level: link:https://access.redhat.com/support/offerings/techpreview[Technology Preview]

> .OpenTelemetry Collector custom resource with configured Health Check extension

Reviewer: ```suggestion
.OpenTelemetry Collector custom resource with the configured Health Check extension
```

[source,yaml]
----
  config: |
    extensions:
      health_check:
        endpoint: "0.0.0.0:13133" <1>
        tls: <2>
          ca_file: "/path/to/ca.crt"
          cert_file: "/path/to/cert.crt"
          key_file: "/path/to/key.key"
        path: "/health/status" <3>
        check_collector_pipeline: <4>
          enabled: true <5>
          interval: "5m" <6>
          exporter_failure_threshold: 5 <7>

    receivers:
      otlp:
        protocols:
          http:

    exporters:
      otlp:

    service:
      extensions: [health_check]
      pipelines:
        traces:
          receivers: [otlp]
          exporters: [otlp]
----

> <1> Address to publish the health check status. Default is `0.0.0.0:13133`.

Reviewer: By `address` you mean an `IP address`?
```suggestion
<1> The target IP address for publishing the health check status. The default is `0.0.0.0:13133`.
```

> <2> The TLS server side configuration. Defines paths to TLS certificates. If omitted, TLS is disabled.

Reviewer: ```suggestion
<2> The TLS server-side configuration. Defines paths to TLS certificates. If omitted, the TLS is disabled.
```

> <3> Specifies the path to be configured for the health check server. Default is `/`.

Reviewer: ```suggestion
<3> The path for the health check server. The default is `/`.
```

> <4> Settings of collector pipeline health check.

Reviewer: ```suggestion
<4> Settings for the Collector pipeline health check.
```

> <5> Enableds the collector pipeline check. default is `false`.

Reviewer: ```suggestion
<5> Enables the Collector pipeline health check. The default is `false`.
```

> <6> Time interval to check the number of failures. Default is `5m`.

Reviewer: ```suggestion
<6> The time interval for checking the number of failures. The default is `5m`.
```

> <7> The failure number threshold to mark containers as healthy. Default is `5`.

Reviewer: ```suggestion
<7> The threshold of a number of failures until which a container is still marked as healthy. The default is `5`.
```

[id="balast-extension_{context}"]

> ==== Balast extension

Reviewer: ```suggestion
[id="memory-ballast-extension_{context}"]
==== Memory Ballast extension
```

> Memory Ballast extension enables applications to configure memory ballast for the process.

Reviewer: ```suggestion
The Memory Ballast extension enables applications to configure memory ballast for the process.
```

* Support level: link:https://access.redhat.com/support/offerings/techpreview[Technology Preview]

> .OpenTelemetry Collector custom resource with configured Balast extension

Reviewer: ```suggestion
.OpenTelemetry Collector custom resource with the configured Memory Ballast extension
```

[source,yaml]
----
  config: |
    extensions:
      memory_ballast:
        size_mib: 64 <1>
        size_in_percentage: 20 <2>

    receivers:
      otlp:
        protocols:
          http:

    exporters:
      otlp:

    service:
      extensions: [memory_ballast]
      pipelines:
        traces:
          receivers: [otlp]
          exporters: [otlp]
----

> <1> Is the memory ballast size, in MiB. Takes higher priority than `size_in_percentage` if both are specified at the same time.

Reviewer: ```suggestion
<1> Sets the memory ballast size in MiB. Takes priority over the `size_in_percentage` if both are specified.
```

> <2> Sets the memory ballast based on the total memory in percentage, value range is 1-100. It is supported in both containerized(eg, docker, k8s) and physical host environments.

Reviewer: ```suggestion
<2> Sets the memory ballast as a percentage, `1`-`100`, of the total memory. Supports containerized and physical host environments.
```

[id="zpages-extension_{context}"]
==== zPages extension

> The zPage extension enables an extension that serves zPages, an HTTP endpoint that provides live data for debugging different components that were properly instrumented for such. All core exporters and receivers provide some zPage instrumentation.

Reviewer: ```suggestion
The zPages extension provides an HTTP endpoint for extensions that serve zPages. At the endpoint, this extension serves live data for debugging instrumented components. All core exporters and receivers provide some zPages instrumentation.
```

> zPages are useful for in-process diagnostics without having to depend on any backend to examine traces or metrics.

Reviewer: ```suggestion
zPages are useful for in-process diagnostics without having to depend on a back end to examine traces or metrics.
```

* Support level: link:https://access.redhat.com/support/offerings/techpreview[Technology Preview]

> .OpenTelemetry Collector custom resource with configured Zpage extension

Reviewer: ```suggestion
.OpenTelemetry Collector custom resource with the configured zPages extension
```

Author: ```suggestion
.OpenTelemetry Collector custom resource with the configured ZPages extension
```

[source,yaml]
----
  config: |
    extensions:
      zpages:
        endpoint: "localhost:55679" <1>

    receivers:
      otlp:
        protocols:
          http:
    exporters:
      otlp:

    service:
      extensions: [zpages]
      pipelines:
        traces:
          receivers: [otlp]
          exporters: [otlp]
----

> <1> Specifies the HTTP endpoint that serves zPages. Use localhost: to make it available only locally, or ":" to make it available on all network interfaces. Default is `localhost:55679`.

Reviewer: ```suggestion
<1> Specifies the HTTP endpoint that serves zPages. Use `localhost:` to make it available only locally, or `":"` to make it available on all network interfaces. The default is `localhost:55679`.
```
